tools/check_format.py

```diff
@@ -164,13 +164,33 @@
 # yapf: enable
 
 
-# Map a line transformation function across each line of a file.
-# .bak temporaries.
-def replaceLines(path, line_xform):
+# Map a line transformation function across each line of a file,
+# writing the result lines as requested.
+# If there is a clang format nesting or mismatch error, return the first occurrence
+def evaluateLines(path, line_xform, write=True):
+  error_message = None
+  format_flag = True
+  output_lines = []
+  for line_number, line in enumerate(readLines(path)):
+    if line.find("// clang-format off") != -1:
+      if not format_flag and error_message is None:
+        error_message = "%s:%d: %s" % (path, line_number + 1, "clang-format nested off")
```

Comment:
as suggested earlier, can you pass reportError as a lambda into evaluateLines so you don't have to repeat the error-message formatting logic here & below? You can still show only one clang-format nesting error if you want by keeping a local bool.

Reply:
I'm going to go ahead and merge this so we don't keep having merge conflicts. @wrowe can you do a follow up to address @jmarantz comments on this file? Thank you.